include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -407,7 +409,8 @@ class FVectorType
 
   /// Strip off a single layer of this type and return the sub-type and a field
   /// ID targeting the same field, but rebased on the sub-type.
-  std::pair<FIRRTLBaseType, size_t> getSubTypeByFieldID(size_t fieldID);
+  std::pair<circt::hw::FieldIDTypeInterface, size_t>
```

Comment:
After we unify these (size_t vs unsigned), it'd be good to wrap them in some named struct or something, also for some type safety perhaps re:how they're used.

Reply:
I just noticed that I guess due how interfaces tie to the concrete operation, that the return type mismatch is able to satisfy the interface method declaration?

(this wouldn't work with other types of inheritance, like normal C++).

I think the interface method should be the larger of the types, so there's no risk of truncation (or getting different values when asking the op directly vs through the interface).